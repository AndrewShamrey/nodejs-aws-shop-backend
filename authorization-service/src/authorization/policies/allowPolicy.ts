import { APIGatewayAuthorizerResult } from 'aws-lambda';

export const allowPolicy = (
  principalId = 'apigateway.amazonaws.com',
  context?: Record<string, string>,
): APIGatewayAuthorizerResult => ({
  principalId,
  context,
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: 'execute-api:Invoke',
        Effect: 'Allow',
        Resource: '*',
      },
    ],
  },
});
