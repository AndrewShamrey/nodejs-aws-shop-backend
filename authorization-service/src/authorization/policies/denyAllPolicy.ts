import { APIGatewayAuthorizerResult } from 'aws-lambda';

export const denyAllPolicy = (): APIGatewayAuthorizerResult => ({
  principalId: '*',
  policyDocument: {
    Version: '2012-10-17',
    Statement: [
      {
        Action: '*',
        Effect: 'Deny',
        Resource: '*',
      },
    ],
  },
});
