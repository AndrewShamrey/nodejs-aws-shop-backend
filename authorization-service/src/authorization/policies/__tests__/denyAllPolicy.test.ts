import { denyAllPolicy } from 'authorization/policies/denyAllPolicy';

describe('denyAllPolicy', () => {
  it('should return valid denyAll policy object', () => {
    expect(denyAllPolicy()).toMatchSnapshot();
  });
});
