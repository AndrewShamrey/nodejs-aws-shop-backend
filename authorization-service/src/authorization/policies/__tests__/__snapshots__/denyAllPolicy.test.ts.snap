// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`denyAllPolicy should return valid denyAll policy object 1`] = `
Object {
  "policyDocument": Object {
    "Statement": Array [
      Object {
        "Action": "*",
        "Effect": "Deny",
        "Resource": "*",
      },
    ],
    "Version": "2012-10-17",
  },
  "principalId": "*",
}
`;
